record/record.controller.js

```diff
@@ -450,6 +456,150 @@
+        vm.deleteRelatedRecord = function (tableModel) {
+            event.preventDefault();
+            event.stopPropagation();
+
+            var params = {};
+            var ref = tableModel.reference;
+
+            // assumption is that this function is only called for p&b
+            params.parentTuple = tableModel.parentTuple
+            params.parentReference = tableModel.parentReference;
+            params.displayMode = recordsetDisplayModes.unlinkPureBinaryPopup;
+
+            params.reference = ref.hideFacets().contextualize.compactSelect; // column reference
+            params.selectMode = modalBox.multiSelectMode;
+            params.selectedRows = [];
+            params.showFaceting = true;
+            params.facetPanelOpen = false;
+
+            var stackElement = logService.getStackNode(
+                logService.logStackTypes.RELATED,
+                params.reference.table,
+                {source: ref.compressedDataSource, entity: true}
+            );
+
+            var logStack = logService.getStackObject(stackElement),
+                logStackPath = logService.getStackPath("", logService.logStackPaths.UNLINK_PB_POPUP);
+
+            params.logStack = logStack;
+            params.logStackPath = logStackPath;
+
+            params.submitBeforeClose = function dataSelected(res, searchPopupTableModel) {
+                // tuples returned are for leaf table, need to get tuples from assocation table instead
+                // if no rows, nothing to delete
+                if (!res || !Array.isArray(res.rows)) return;
+                var tuples = res.rows;
+
+                var CONFIRM_DELETE = (chaiseConfig.confirmDelete === undefined || chaiseConfig.confirmDelete) ? true : false;
+                // NOTE: This reference has to be filtered so creating the path in the ermrestJS function works properly
+                var leafReference = tuples[0].reference;
+                $rootScope.showSpinner = true;
+
+                // success response function for deleteBatchAssociationTuples
+                // response is an error object with message for both success and error info
+                var deleteResponse = function (response) {
+                    $rootScope.showSpinner = false;
+
+                    // Show modal popup summarizing total # of deletions succeeded and failed
+                    response.clickOkToDismiss = true;
+                    var reloadModalRows = function () {
+                        if (response.failedTupleData.length > 0) {
+                            // iterate over the set of successful ids and find them in selected rows, then remove them
+                            response.successTupleData.forEach(function (data) {
+                                // data is an object of key/value pairs for each piece of key information
+                                // { keycol1: val, keycol2: val2, ... }
+                                // TODO: what to do with multiple keys?
```

Comment:
You should handle composite keys. You just need to change your `findIndex` to call a `every` (or `some`) on all the key values. Something like this

```js
searchPopupTableModel.selectedRows.findIndex(function (tuple) {
  return Object.keys(data).every(function (key) {
    return data[key] == tuple[key];
  });
});
```

And to be honest given that you're passing the `selectedRows` to the function, you could just return the actual given `Tuple` object without any modification. Then you could either do object comparison or rely on `Tuple.uniqueId` API for comparison. That being said, what you're passing from ermrestjs to chaise is good enough for now.